### PR TITLE
Make JAX compatible with numpy version 1.23.0

### LINKED ISSUE #11221: autodiff fails under numpy 1.23
Something in the numpy 1.23 release broke autodiff; in particular, float0 arrays are being converted to bool:
```python
import jax
import numpy
print(numpy.__version__)
print(jax.grad(lambda x: x+0., allow_int=True)(1).dtype)
```
Newest numpy:
```
1.23.0
bool
```
previous version:
```
1.22.4
[('float0', 'V')]
```

### PR DESCRIPTION
**Pull request by jakevdp**
For more discussion of these changes, see #11221

Fixes #11221

### CODE REVIEW

**Comment by jakevdp** [comment]
(related: https://github.com/numpy/numpy/issues/21838 and https://github.com/numpy/numpy/issues/21840)

File: build/test-requirements.txt [hunk 1]
```diff
@@ -6,4 +6,4 @@ pillow>=8.3.1,<9.1.0
 pytest-benchmark
 pytest-xdist
 wheel
-numpy<1.23.0
+numpy
```
Comment (yashk2810) [inline build/test-requirements.txt:9]: Remove this? setup.py should cover it

**Review by yashk2810 — APPROVED** [review]

File: build/test-requirements.txt (hunk 1 above)
Comment (jakevdp) [reply build/test-requirements.txt:9]: I don't understand - we want the tests to run on the most recent version of numpy.

Comment (yashk2810) [reply build/test-requirements.txt:9]: Right. There is no upper bound on version constraint on numpy.

This numpy was not there before I added the `<1.23.0` and we still hit that error. So I think removing this should be fine.

**Pull request merged**